File: src/ClientApp/src/app/pages/clients/view-models/client-product.view-model.ts
export interface ClientProductViewModel {
    id: string;
    clientId: string;
    productId: string;
    productCode: string;
    productName: string;
    relationshipStatus: string;
    opportunityId?: string | null;
    opportunityNumber?: string | null;
    opportunityTitle?: string | null;
    ownerUserId?: string | null;
    ownerUserName?: string | null;
    startDate?: string | null;
    endDate?: string | null;
    notes?: string | null;
}

File: src/API/src/Modules/Clients/Clients.Application/ViewModels/ClientProductViewModel.cs
namespace Clients.Application.ViewModels
{
    public class ClientProductViewModel
    {
        public Guid Id { get; set; }
        public Guid ClientId { get; set; }
        public Guid ProductId { get; set; }
        public string ProductCode { get; set; } = string.Empty;
        public string ProductName { get; set; } = string.Empty;
        public string RelationshipStatus { get; set; } = string.Empty;
        public Guid? OpportunityId { get; set; }
        public string? OpportunityNumber { get; set; }
        public string? OpportunityTitle { get; set; }
        public Guid? OwnerUserId { get; set; }
        public string? OwnerUserName { get; set; }
        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public string? Notes { get; set; }
    }
}
